packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,209 @@
+      },
+    });
+
+    this.functionArn = customResource.getResponseField('Parameter.Value');
+    // this.lambda = lambda.Function.fromFunctionArn(this, 'ImportedFn', this.functionArn);
+    this.lambda = edgeFunction;
+    this.functionName = this.lambda.functionName;
+    this.grantPrincipal = this.lambda.role!; // TODO - Does this work, or does it need to be imported?
+    this.permissionsNode = this.lambda.permissionsNode;
+    this.edgeArn = this.functionArn;
+    this.version = lambda.extractQualifierFromArn(this.functionArn);
+  }
+
+  /**
+   * Not supported. Connections are only applicable to VPC-enabled functions.
+   */
+  public get connections(): ec2.Connections {
+    throw new Error('Lambda@Edge does not support connections');
+  }
+  public get latestVersion(): lambda.IVersion {
+    throw new Error('$LATEST function version cannot be used for Lambda@Edge');
+  }
+
+  public addEventSourceMapping(id: string, options: lambda.EventSourceMappingOptions): lambda.EventSourceMapping {
+    return this.lambda.addEventSourceMapping(id, options);
+  }
+  public addPermission(id: string, permission: lambda.Permission): void {
+    return this.lambda.addPermission(id, permission);
+  }
+  public addToRolePolicy(statement: iam.PolicyStatement): void {
+    return this.lambda.addToRolePolicy(statement);
+  }
+  public grantInvoke(identity: iam.IGrantable): iam.Grant {
+    return this.lambda.grantInvoke(identity);
+  }
+  public metric(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metric(metricName, { ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricDuration(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricDuration({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricErrors({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricInvocations(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricInvocations({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricThrottles(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricThrottles({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public addEventSource(source: lambda.IEventSource): void {
+    return this.lambda.addEventSource(source);
+  }
+  public configureAsyncInvoke(options: lambda.EventInvokeConfigOptions): void {
+    return this.lambda.configureAsyncInvoke(options);
+  }
+
+  public addAlias(aliasName: string, options: lambda.AliasOptions = {}): lambda.Alias {
+    return new lambda.Alias(this.functionStack, `Alias${aliasName}`, {
+      aliasName,
+      version: this.currentVersion,
+      ...options,
+    });
+  }
+
+  private getOrCreateEdgeStack(): CrossRegionLambdaStack {
+    const app = this.node.root;
+    if (!app || !App.isApp(app)) {
```

Comment:
Why do we need this check?

Also, I think this should be `Stage.isStage()`.

Reply:
The app is used down on line 155 to lookup if the supporting stack already exists within the app. This is stolen from: https://github.com/aws/aws-cdk/blob/b5f2f75128e156c04bcd4b4f3513d7387515be1d/packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts#L491-L493

Why `Stage` vs `App` in this context?

Reply:
In the case of CDK pipelines, multiple instances of the stack will exist per stage.